tbf/testcase_validation.py

```diff
@@ -390,19 +390,28 @@ def check_inputs(self,
                                               test_vector, nondet_methods)
                 with open(harness['name'], 'wb+') as outp:
                     outp.write(harness['content'])
+
                 result.harness = harness['name']
 
+            # Create an ExecutionRunner only for the purpose of
+            # compiling the final harness
+            validator = ExecutionRunner(self.config.machine_model,
+                                        self.get_name(),
+                                        harness_file=result.harness,
+                                        save_exec=True)
+            validator.compile(program_file, result.harness)
         return result, self.statistics
 
 
 class ExecutionRunner(object):
 
-    def __init__(self, machine_model, producer_name):
+    def __init__(self, machine_model, producer_name, harness_file="harness.c", save_exec=False):
         self.machine_model = machine_model
         self.harness = None
         self.producer = producer_name
         self.harness_generator = harness_gen.HarnessCreator()
-        self.harness_file = 'harness.c'
+        self.harness_file = harness_file
```

Comment:
There is no directory, so `harness.c` will resolve to the one in the temporary directory.  But I want to compile `harness.c` that is in the `output` directory instead.

Reply:
Is this actually working?
I thought that `self.harness_file` is only used to decide where to write a newly created harness when calling `_create_executable_harness`:
https://github.com/sosy-lab/tbf/blob/5ab9374e0e91e3fda8a768927643d63623549696/tbf/testcase_validation.py#L447-L453

The persistent harness should only be copied from the temporary directory to the `output` directory
at the end of the main procedure:
https://github.com/sosy-lab/tbf/blob/5ab9374e0e91e3fda8a768927643d63623549696/tbf/__init__.py#L375-L382

Reply:
Oops, I was looking at this more and you're right.  I can just pass the new path to compile.

Reply:
It is working.  The harness is compiled from `/tmp` (e.g., `/tmp/tmp6u8s1ez7/klee-tests/test000002.ktest.harness.c`).  I didn't know about the code above in __init__.py, so that is happening afterward.

Maybe it would make more sense to compile the persistent harness in __init__.py.

Reply:
I think your concerns about the ExecutionRunner arguments should be resolved now.

Reply:
Yeah, I think you're right - this loop in `__init__.py` is actually quite ugly and complicates post-processing:

https://github.com/sosy-lab/tbf/blob/5ab9374e0e91e3fda8a768927643d63623549696/tbf/__init__.py#L372-L373

Should I quickly split this to two `if`-statements (one to check whether there's a harness, one to check whether there's a witness), or do you want to do it yourself/don't care about the current structure?

Reply:
Go ahead and split it up and then I'll pull and try to move the compilation of the harness there.

Reply:
I just pushed the change.

Reply:
Me too!